modules/teacher.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission: parent_permission)
+    @specialization = specialization
+  end
```

Comment:
As mentioned above and in order to avoid errors, kindly call the constructor of the parent class passing the parameters in the same order that they appear (age, name, parent_permission).

Reply:
Thanks for the feedback.
The changes have been implemented.

Reply:
Fixed.